samples: Bluetooth: df_rx: Add info about CTE TX/RX configuration (#5187)
nrfconnect/sdk-nrf: samples/bluetooth/direction_finding_connectionless_rx/README.rst

There are two antenna switch slot length allowed: 1 us or 2 us.
The transmitter uses the antenna switch slot length to configure radio peripheral in Angle of Deprature mode only.
The receiver uses the local setting of antenna switch slot length in Angle of Arrival mode only.
When CTE type in received PDU is AoD, the receivers radio peripheral uses the antenna switch slot length appropriate for AoD type: 1 us or 2 us.
The antenna switch slot length has influece on number of IQ samples provided in IQ samples report.

Constant Tone Extension length is limited to range 16 us to 160 us, the value is provided in 8 us units.
The transmitter is responsible for setting of CTE length.
The value is send to receiver as part of CTEInfo field in PDUs extended advertising header.
The receivers radio peripheral uses the CTE length, provided in periodic advertising PDU, to execute antenna switching and CTE sampling in AoA mode or CTE sampling only in AoD mode.
The CTE length has influence on number of IQ samples provided in IQ samples report.

Constant Tone Extension consists of three periods:

> * Guard period that lasts for 4 us. It is a gap before actual CTE reception and adjacent PDU transmission to avoid interference.

Reviewer: ```suggestion
* Guard period that lasts for 4 µs. There is a gap before the actual CTE reception and the adjacent PDU transmission to avoid interference.
```

> * Reference period where single antenna is used sampling, samples are spaced each other by 1 us. The period duration is 8 us.

Reviewer: ```suggestion
* Reference period where single antenna is used for sampling. The samples are spaced 1 µs from each other. The period duration is 8 µs.
```

> * Switch-sample period that is split into switch and sample slots. Every slot may either be 1 us or 2 us long.

Reviewer: ```suggestion
* Switch-sample period that is split into switch and sample slots. Each slot may be either 1 µs or 2 µs long.
```

> Total number of IQ samples provided by IQ samples report varies.

Reviewer: ```suggestion
The total number of IQ samples provided by the IQ samples report varies.
```

It depends on the CTE length and the antenna switch slot length.

> There will always be 8 samples from reference period, 1 to 37 samples with 2 us atnenna switching slots, 2 to 74 samples with 1 us antenna switching slots, meaning 9 to 82 samples in total.

Reviewer: ```suggestion
There will always be 8 samples from the reference period, 1 to 37 samples with 2 µs antenna switching slots, 2 to 74 samples with 1 µs antenna switching slots, meaning 9 to 82 samples in total.
```

> For example CTE length is 120 us, antenna switching slot is 1 us.

Reviewer: ```suggestion
For example, the CTE length is 120 µs and the antenna switching slot is 1 µs.
```

> There are 8 samples from reference period.
> Switch sample period duration is: 120 us - 16 us = 104 us.
> For antenna switching slot 1 us, sample is taken every 2 us (1 us antenna switch slot, 1 us sample slot).
> Number of samples from the switch-sample period is 104 us / 2 us = 52.
> Total numnber of samples is 60.

Reviewer: ```suggestion
This will result in the following number of IQ samples:

* 8 samples from the reference period.
* Switch sample period duration is: 120 µs - 16 µs = 104 µs. For the 1 µs antenna switching slot, the sample is taken every 2 µs (1 µs antenna switch slot, 1 µs sample slot). Meaning, the number of samples from the switch-sample period is 104 µs / 2 µs = 52.

The total number of samples is 60.
```

The CTE length has influence on number of IQ samples provided in the IQ samples report.

Constant Tone Extension consists of three periods:

* Guard period that lasts for 4 µs. There is a gap before the actual CTE reception and the adjacent PDU transmission to avoid interference.
* Reference period where single antenna is used for sampling. The samples are spaced 1 µs from each other. The period duration is 8 µs.
* Switch-sample period that is split into switch and sample slots. Each slot may be either 1 µs or 2 µs long.

The total number of IQ samples provided by the IQ samples report varies.
It depends on the CTE length and the antenna switch slot length.
There will always be 8 samples from the reference period, 1 to 37 samples with 2 µs antenna switching slots, 2 to 74 samples with 1 µs antenna switching slots, meaning 9 to 82 samples in total.

For example, the CTE length is 120 µs and the antenna switching slot is 1 µs.
This will result in the following number of IQ samples:

* 8 samples from the reference period.

> * Switch sample period duration is: 120 µs - 16 µs = 104 µs. For the 1 µs antenna switching slot, the sample is taken every 2 µs (1 µs antenna switch slot, 1 µs sample slot). Meaning, the number of samples from the switch-sample period is 104 µs / 2 µs = 52.

Reviewer: Switch sample period -> Switch-sample period

Just to align with other occurrences :)

Author: Corrected :-)
